File: android_stuff/bitfountain/DoggyDogWorld/app/src/main/java/com/nocdib/doggydogworld/Dog.java
package com.nocdib.doggydogworld;


public class Dog {

    private String  name,
            breed;
    private int age;

    public Dog(){
        name = "Fido";
        breed = "Mutt";
        age = 2;
    }

    public Dog(String name, String breed, int age){
        this.name = name;
        this.breed = breed;
        this.age = age;
    }

    public String getName(){
        return name;
    }

    public String getBreed(){
        return breed;
    }

    public int getAge(){
        return age;
    }

    public String bark(){
        return "Woof! Woof!";
    }

}

File: js_stuff/module15/dog.js
class Dog {

    constructor(name, breed){
        this.name = name;
        this.breed = breed;
    }

    bark(){
        console.log(`Bark! My name is ${this.name}`);
    }
}

const renley = new Dog('Renly', 'Beagle');

console.log(renley);
renley.bark();